### ISSUE TITLE
Postgres procedure missing WITH clause option

### ISSUE THREAD

**Issue by PavanIntegra** [sqlfluff/sqlfluff#6481]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The postgres dialect does not handle `WITH` clause for procedures. 
This query fails linting
```
CREATE PROCEDURE abc.cdf()
LANGUAGE sql
BEGIN ATOMIC
WITH tbl2 AS (
SELECT a.apple
FROM tbl1 a
)

SELECT t.apple
FROM tbl2 t
;
END;
```
`Found unparsable section: 'BEGIN | ATOMIC\nWITH tbl1 AS (\nSELECT`

If a query without a `WITH` clause is used, the linter works
```
CREATE PROCEDURE abc.cdf()
LANGUAGE sql
BEGIN ATOMIC

SELECT t.apple
FROM tbl1 t
;
END;
```

I was able to fix this by adding this to https://github.com/sqlfluff/sqlfluff/blob/83a404a7eb76a1f6612c44ea1f0ea8cd1e90885c/src/sqlfluff/dialects/dialect_postgres.py#L1596
```
                    Sequence(Ref("WithCompoundStatementSegment"),
                        Ref("SemicolonSegment"),),
```

This may not be the correct fix but it did work for me.

### Expected Behaviour

```
CREATE PROCEDURE abc.cdf()
LANGUAGE sql
BEGIN ATOMIC
WITH tbl2 AS (
SELECT a.apple
FROM tbl1 a
)

SELECT t.apple
FROM tbl2 t
;
END;
```

Should return
```
INFO       
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        create_procedure_statement:
[L:  1, P:  1]      |            keyword:                                          'CREATE'
[L:  1, P:  7]      |            whitespace:                                       ' '
[L:  1, P:  8]      |            keyword:                                          'PROCEDURE'
[L:  1, P: 17]      |            whitespace:                                       ' '
[L:  1, P: 18]      |            function_name:
[L:  1, P: 18]      |                naked_identifier:                             'abc'
[L:  1, P: 21]      |                dot:                                          '.'
[L:  1, P: 22]      |                function_name_identifier:                     'cdf'
[L:  1, P: 25]      |            function_parameter_list:
[L:  1, P: 25]      |                bracketed:
[L:  1, P: 25]      |                    start_bracket:                            '('
[L:  1, P: 26]      |                    [META] indent:
[L:  1, P: 26]      |                    [META] dedent:
[L:  1, P: 26]      |                    end_bracket:                              ')'
[L:  1, P: 27]      |            newline:                                          '\n'
[L:  2, P:  1]      |            function_definition:
[L:  2, P:  1]      |                language_clause:
[L:  2, P:  1]      |                    keyword:                                  'LANGUAGE'
[L:  2, P:  9]      |                    whitespace:                               ' '
[L:  2, P: 10]      |                    naked_identifier:                         'sql'
[L:  2, P: 13]      |                newline:                                      '\n'
[L:  3, P:  1]      |                keyword:                                      'BEGIN'
[L:  3, P:  6]      |                whitespace:                                   ' '
[L:  3, P:  7]      |                keyword:                                      'ATOMIC'
[L:  3, P: 13]      |                newline:                                      '\n'
[L:  4, P:  1]      |                with_compound_statement:
[L:  4, P:  1]      |                    keyword:                                  'WITH'
[L:  4, P:  5]      |                    whitespace:                               ' '
[L:  4, P:  6]      |                    common_table_expression:
[L:  4, P:  6]      |                        naked_identifier:                     'tbl2'
[L:  4, P: 10]      |                        whitespace:                           ' '
[L:  4, P: 11]      |                        keyword:                              'AS'
[L:  4, P: 13]      |                        whitespace:                           ' '
[L:  4, P: 14]      |                        bracketed:
[L:  4, P: 14]      |                            start_bracket:                    '('
[L:  4, P: 15]      |                            [META] indent:
[L:  4, P: 15]      |                            newline:                          '\n'
[L:  5, P:  1]      |                            select_statement:
[L:  5, P:  1]      |                                select_clause:
[L:  5, P:  1]      |                                    keyword:                  'SELECT'
[L:  5, P:  7]      |                                    [META] indent:
[L:  5, P:  7]      |                                    whitespace:               ' '
[L:  5, P:  8]      |                                    select_clause_element:
[L:  5, P:  8]      |                                        column_reference:
[L:  5, P:  8]      |                                            naked_identifier:  'a'
[L:  5, P:  9]      |                                            dot:              '.'
[L:  5, P: 10]      |                                            naked_identifier:  'apple'
[L:  5, P: 15]      |                                    [META] dedent:
[L:  5, P: 15]      |                                newline:                      '\n'
[L:  6, P:  1]      |                                from_clause:
[L:  6, P:  1]      |                                    keyword:                  'FROM'
[L:  6, P:  5]      |                                    whitespace:               ' '
[L:  6, P:  6]      |                                    from_expression:
[L:  6, P:  6]      |                                        [META] indent:
[L:  6, P:  6]      |                                        from_expression_element:
[L:  6, P:  6]      |                                            table_expression:
[L:  6, P:  6]      |                                                table_reference:
[L:  6, P:  6]      |                                                    naked_identifier:  'tbl1'
[L:  6, P: 10]      |                                            whitespace:       ' '
[L:  6, P: 11]      |                                            alias_expression:
[L:  6, P: 11]      |                                                naked_identifier:  'a'
[L:  6, P: 12]      |                                        [META] dedent:
[L:  6, P: 12]      |                            newline:                          '\n'
[L:  7, P:  1]      |                            [META] dedent:
[L:  7, P:  1]      |                            end_bracket:                      ')'
[L:  7, P:  2]      |                    newline:                                  '\n'
[L:  8, P:  1]      |                    newline:                                  '\n'
[L:  9, P:  1]      |                    select_statement:
[L:  9, P:  1]      |                        select_clause:
[L:  9, P:  1]      |                            keyword:                          'SELECT'
[L:  9, P:  7]      |                            [META] indent:
[L:  9, P:  7]      |                            whitespace:                       ' '
[L:  9, P:  8]      |                            select_clause_element:
[L:  9, P:  8]      |                                column_reference:
[L:  9, P:  8]      |                                    naked_identifier:         't'
[L:  9, P:  9]      |                                    dot:                      '.'
[L:  9, P: 10]      |                                    naked_identifier:         'apple'
[L:  9, P: 15]      |                            [META] dedent:
[L:  9, P: 15]      |                        newline:                              '\n'
[L: 10, P:  1]      |                        from_clause:
[L: 10, P:  1]      |                            keyword:                          'FROM'
[L: 10, P:  5]      |                            whitespace:                       ' '
[L: 10, P:  6]      |                            from_expression:
[L: 10, P:  6]      |                                [META] indent:
[L: 10, P:  6]      |                                from_expression_element:
[L: 10, P:  6]      |                                    table_expression:
[L: 10, P:  6]      |                                        table_reference:
[L: 10, P:  6]      |                                            naked_identifier:  'tbl2'
[L: 10, P: 10]      |                                    whitespace:               ' '
[L: 10, P: 11]      |                                    alias_expression:
[L: 10, P: 11]      |                                        naked_identifier:     't'
[L: 10, P: 12]      |                                [META] dedent:
[L: 10, P: 12]      |                newline:                                      '\n'
[L: 11, P:  1]      |                statement_terminator:                         ';'
[L: 11, P:  2]      |                newline:                                      '\n'
[L: 12, P:  1]      |                keyword:                                      'END'
[L: 12, P:  4]      |    statement_terminator:                                     ';'
[L: 12, P:  5]      |    newline:                                                  '\n'
[L: 13, P:  1]      |    [META] end_of_file:

== [/TEST.sql] PASS
```

### Observed Behaviour

Instead it is returning
```
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        create_procedure_statement:
[L:  1, P:  1]      |            keyword:                                          'CREATE'
[L:  1, P:  7]      |            whitespace:                                       ' '
[L:  1, P:  8]      |            keyword:                                          'PROCEDURE'
[L:  1, P: 17]      |            whitespace:                                       ' '
[L:  1, P: 18]      |            function_name:
[L:  1, P: 18]      |                naked_identifier:                             'abc'
[L:  1, P: 21]      |                dot:                                          '.'
[L:  1, P: 22]      |                function_name_identifier:                     'cdf'
[L:  1, P: 25]      |            function_parameter_list:
[L:  1, P: 25]      |                bracketed:
[L:  1, P: 25]      |                    start_bracket:                            '('
[L:  1, P: 26]      |                    [META] indent:
[L:  1, P: 26]      |                    [META] dedent:
[L:  1, P: 26]      |                    end_bracket:                              ')'
[L:  1, P: 27]      |            newline:                                          '\n'
[L:  2, P:  1]      |            function_definition:
[L:  2, P:  1]      |                language_clause:
[L:  2, P:  1]      |                    keyword:                                  'LANGUAGE'
[L:  2, P:  9]      |                    whitespace:                               ' '
[L:  2, P: 10]      |                    naked_identifier:                         'sql'
[L:  2, P: 13]      |    newline:                                                  '\n'
[L:  3, P:  1]      |    unparsable:                                               !! Expected: 'Nothing else in FileSegment.'
[L:  3, P:  1]      |        word:                                                 'BEGIN'
[L:  3, P:  6]      |        whitespace:                                           ' '
[L:  3, P:  7]      |        word:                                                 'ATOMIC'
[L:  3, P: 13]      |        newline:                                              '\n'
[L:  4, P:  1]      |        word:                                                 'WITH'
[L:  4, P:  5]      |        whitespace:                                           ' '
[L:  4, P:  6]      |        word:                                                 'tbl2'
[L:  4, P: 10]      |        whitespace:                                           ' '
[L:  4, P: 11]      |        word:                                                 'AS'
[L:  4, P: 13]      |        whitespace:                                           ' '
[L:  4, P: 14]      |        start_bracket:                                        '('
[L:  4, P: 15]      |        newline:                                              '\n'
[L:  5, P:  1]      |        word:                                                 'SELECT'
[L:  5, P:  7]      |        whitespace:                                           ' '
[L:  5, P:  8]      |        word:                                                 'a'
[L:  5, P:  9]      |        dot:                                                  '.'
[L:  5, P: 10]      |        word:                                                 'apple'
[L:  5, P: 15]      |        newline:                                              '\n'
[L:  6, P:  1]      |        word:                                                 'FROM'
[L:  6, P:  5]      |        whitespace:                                           ' '
[L:  6, P:  6]      |        word:                                                 'tbl1'
[L:  6, P: 10]      |        whitespace:                                           ' '
[L:  6, P: 11]      |        word:                                                 'a'
[L:  6, P: 12]      |        newline:                                              '\n'
[L:  7, P:  1]      |        end_bracket:                                          ')'
[L:  7, P:  2]      |        newline:                                              '\n'
[L:  8, P:  1]      |        newline:                                              '\n'
[L:  9, P:  1]      |        word:                                                 'SELECT'
[L:  9, P:  7]      |        whitespace:                                           ' '
[L:  9, P:  8]      |        word:                                                 't'
[L:  9, P:  9]      |        dot:                                                  '.'
[L:  9, P: 10]      |        word:                                                 'apple'
[L:  9, P: 15]      |        newline:                                              '\n'
[L: 10, P:  1]      |        word:                                                 'FROM'
[L: 10, P:  5]      |        whitespace:                                           ' '
[L: 10, P:  6]      |        word:                                                 'tbl2'
[L: 10, P: 10]      |        whitespace:                                           ' '
[L: 10, P: 11]      |        word:                                                 't'
[L: 10, P: 12]      |        newline:                                              '\n'
[L: 11, P:  1]      |        semicolon:                                            ';'
[L: 11, P:  2]      |        newline:                                              '\n'
[L: 12, P:  1]      |        word:                                                 'END'
[L: 12, P:  4]      |        semicolon:                                            ';'
[L: 12, P:  5]      |    newline:                                                  '\n'
[L: 13, P:  1]      |    [META] end_of_file:

== [/TEST.sql] FAIL
```

### How to reproduce

```
CREATE PROCEDURE abc.cdf()
LANGUAGE sql
BEGIN ATOMIC
WITH tbl2 AS (
SELECT a.apple
FROM tbl1 a
)

SELECT t.apple
FROM tbl2 t
;
END;
```

### Dialect

postgres

### Version

sqlfluff, version 3.2.5
python: 3.11.3

### Configuration

No configuration

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by joaostorrer** [comment]
Your fix is correct, but it can be also fixed with
```python
OneOf(Ref("WithCompoundStatementSegment"), Ref("SelectStatementSegment")),
```
Are you willing to submit a PR to fix this issue?
